src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -426,6 +429,8 @@ bool ConnectivityManagerImpl::_IsWiFiStationProvisioned()
     bool ret          = false;
     const gchar * bss = nullptr;
 
+    std::lock_guard<std::mutex> lock(mWpaSupplicantMutex);
```

Comment:
@mspang based on our updated approach of 'external synchronization', where does an object like this under platform/* fall in that spectrum?

Reply:
We should remove the 2nd thread, which I think I said before even that discussion even happened and asked Kevin to help us make that possible by modernizing the event loop.

Reply:
Is the other thread the `std::thread wifiIPThread(WiFIIPChangeListener);`?

That also has the problem of being leaked every time chip starts.  See https://github.com/project-chip/connectedhomeip/issues/9212.

Reply:
Is there an issue filed for removing that thread, and is @kpschoedel working on that? If so, maybe we can let this PR go, and have it be fixed later by Kevin when he re-factors that?